src/main/webapp/assignPanel.js

```diff
@@ -90,24 +90,17 @@ function parseEmailIdFromTaskNotes(task) {
     return null;
   }
 
-  let index = notes.length;
-  let emailIdValue = '';
-  while
-  (emailIdValue.substr(0, 'emailId:'.length) !== 'emailId:' && index > 0) {
-    emailIdValue = notes.charAt(index-1) + emailIdValue;
-    index--;
-  }
-  if (index < 0) {
+  const emailIdIndex = notes.lastIndexOf('emailId:');
+
+  if (emailIdIndex === -1) {
     console.log('No email ID found!');
     return null;
   }
-  return emailIdValue.split(':')[1];
+  return notes.substr(emailIdIndex).split(':')[1];
```

Comment:
nit: just `notes.substr(emailIdIndex + 'emailId:'.length)`?

Reply:
Yes that would work! Thanks!